### PR TITLE
Add dynamic transform matmul

### PR DESCRIPTION
**Pull request by JonasCrols**
Create Automatically tiled matmul example where matrices can be of any size as long as it is a multiple of 8.
fixed Bug where compiler assumes contiguous blocks in TSL. Solved by providing striding-information at runtime.
Added benchmarks

### CODE REVIEW

**Comment by jorendumoulin** [comment]
Thanks! I'll quickly clean up the mess i wrote 😅

**Comment by jorendumoulin** [comment]
I commited my changes over an hour ago but they don't seem to appear here? 🤔

File: kernels/streamer_matmul/main.c [hunk 1]
```diff
@@ -28,10 +28,14 @@
...
-uint8_t M_param = M_size / meshRow;
+
+/* M_param and N_param can be set to 1 for tiled versions, but not for simple
+ version. 2 always works. however, it will impact performance significantly as
+ computation cost doubles. For benchmarks, set to 1 */
+uint8_t M_param = 2;
 uint8_t K_param = K_size / tileSize;
-uint8_t N_param = N_size / meshCol;
+uint8_t N_param = 2;
+
```
Comment (jorendumoulin) [inline kernels/streamer_matmul/main.c:38]: This is a bit fishy but should get resolved when we compile for this kernel from within MLIR!

**Review by jorendumoulin — APPROVED** [review]
Lovely!

**Comment by JosseVanDelm** [comment]
Great work guys! :partying_face:

**Pull request merged**